### PR TITLE
#7 - jbrowse file selection logic

### PR DESCRIPTION
**Pull request by samrichca**
- query arranger to see if files in selected table rows are compatible with jbrowse
- update tooltip text
- reorganize files, make a jbrowse folder
- no changes to dms-jbrowse, use develop branch for testing

### CODE REVIEW

File: components/pages/explorer/Jbrowse/JbrowseLaunchButton.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      unmountHTMLWhenHide
+      arrow
+      html={
+        <div
+          css={css`
+            ${theme.typography.regular};
+            font-size: 12px;
+          `}
+        >
+          Please select 1 to {MAX_JBROWSE_FILES} files to launch JBrowse.
```
Comment (samrichca) [inline components/pages/explorer/Jbrowse/JbrowseLaunchButton.tsx:103]: verbiage change confirmed with @b-f-chan

File: components/pages/explorer/Jbrowse/JbrowseLaunchButton.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import { css, useTheme } from '@emotion/react';
+import { ButtonWrapper } from '../ActionBar';
+import Button from '@/components/Button';
+import { RepositoryTabNames } from '../RepositoryContent';
+import { find } from 'lodash';
+import createArrangerFetcher from '@/components/utils/arrangerFetcher';
+import { useEffect, useState } from 'react';
+import SQON from '@overture-stack/sqon-builder';
+import { JbrowseButtonQueryNode } from './types';
+import { jbrowseAllowedFileTypes, jbrowseFileMetadataQuery, MAX_JBROWSE_FILES } from './utils';
```
[the remark this answers is not in the document]
Comment (samrichca) [reply components/pages/explorer/Jbrowse/JbrowseLaunchButton.tsx:34]: i'm down for automating this (not right now 😂) but not doing it manually.

File: components/pages/explorer/Jbrowse/JbrowseWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        margin-top: 8px;
+        .MuiPaper-elevation12 {
+          // elevation in MUI controls drop shadow
+          box-shadow: none;
+        }
+      `}
+    >
+      <JbrowseLinear
+        options={{
+          configuration: {
```
Comment (justincorrigible) [inline components/pages/explorer/Jbrowse/JbrowseWrapper.tsx:40]: Not a change request, but more of a general consult: other than `configuration`, what else is there in `options`?
The two words sound synonymous to me, so I wonder whether a simpler approach can be reached for `JBrowseLinear`'s props

File: package.json [hunk 4]
```diff
@@ -28,6 +29,7 @@
 		"react-dom": "^17.0.2",
 		"react-grid-system": "^7.1.1",
 		"react-tippy": "^1.4.0",
+		"sqon-builder": "^2.0.1",
```
Comment (justincorrigible) [inline package.json:32]: looks like a duplicate?

File: package.json [hunk 5]
```diff
@@ -50,7 +52,7 @@
 		"trailingComma": "all",
 		"semi": true,
 		"singleQuote": true,
-		"useTabs": true
+		"useTabs": false
```
Comment (justincorrigible) [inline package.json:55]: oops! 😆 
```suggestion
		"useTabs": true
```

File: components/pages/explorer/Jbrowse/JbrowseLaunchButton.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (justincorrigible) [reply components/pages/explorer/Jbrowse/JbrowseLaunchButton.tsx:34]: @joneubank do you know of a way to automate this? generally I just add them in an organised manner when I create new files, so I'm not sure how to make prettier apply ESLint changes upon file saving.

File: package.json (hunk 5 above)
Comment (samrichca) [reply package.json:55]: i made a WIP PR for this feature branch with a note to change this back

File: components/pages/explorer/Jbrowse/JbrowseWrapper.tsx (hunk 3 above)
Comment (samrichca) [reply components/pages/explorer/Jbrowse/JbrowseWrapper.tsx:40]: quite a bit

```
export const defaultCircularOptions = {
  configuration: {},
  connections: [],
  defaultSession: {
    name: "My session",
    view: {
      id: "circularView",
      type: "CircularView",
    },
  },
  tracks: [],
};
```

File: package.json (hunk 5 above)
Comment (justincorrigible) [reply package.json:55]: can't we just not commit this `false`?

File: components/pages/explorer/Jbrowse/JbrowseWrapper.tsx (hunk 3 above)
Comment (justincorrigible) [reply components/pages/explorer/Jbrowse/JbrowseWrapper.tsx:40]: uh huh! indeed! wonder whether those could be spread into separate props: `configuration`, `connections`, etc.

`options.configuration` sounds redundant like `data.info`

File: components/pages/explorer/Jbrowse/JbrowseLaunchButton.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (justincorrigible) [reply components/pages/explorer/Jbrowse/JbrowseLaunchButton.tsx:34]: As discussed in chat, I'm going to look into a VSCode plugin solution (or local/dev-build pipeline approach) for these to be handled.

File: components/pages/explorer/Jbrowse/JbrowseWrapper.tsx (hunk 3 above)
Comment (samrichca) [reply components/pages/explorer/Jbrowse/JbrowseWrapper.tsx]: updated

**Pull request merged**